### PR TITLE
Basic styling for "Application Summary" page

### PR DESCRIPTION
**Pull request by ssandbekkhaug**
### Description

Add styling for questions and answers on the summary page. [View mocks here](https://www.figma.com/design/RmhVJCLgAKWGENJLWWAtNS/CiviForm%3A-Components%2C-Styles?node-id=2459-23883&m=dev).

* Improve alignment
* Add horizontal separator lines
* Adjust margins
* Update fonts (e.g. make answer text darker)
* Truncate questions and answers to 3 lines on mobile
* Removes unnecessary aria-label (The aria-label was mistakenly added by me in a previous PR. It reads "opens in a new window" even though it's not a link.)

#### Before

Summary Before

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).

#### User visible changes

- [x] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [x] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Made equivalent changes in Thymeleaf for applicant-facing features and changes (or created an issue in the [Northstar epic](https://github.com/orgs/civiform/projects/1/views/94) to track that it's missing from Thymeleaf)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order
- [x] Manually tested with a screen reader if the feature is applicant-facing. See [screen reader testing](https://github.com/civiform/civiform/wiki/Testing#screen-reader-testing)

### Issue(s) this completes

Part of #8794

### CODE REVIEW

**Comment by ssandbekkhaug** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/civiform/civiform/8889?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8889** <a href="https://app.graphite.dev/github/pr/civiform/civiform/8889?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8855** <a href="https://app.graphite.dev/github/pr/civiform/civiform/8855?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ssandbekkhaug and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss [hunk 1]
```diff
@@ -140,6 +148,15 @@ it is around the label */
   margin-top: 0;
 }
 
+// This should only be used to override styling.
+// Prefer using class="usa-prose" in parent element where possible.
+.cf-prose-h3 {
+  font-family: var(--heading, Merriweather);
+  @include u-font_size('heading', 'lg');
```
Comment (dkatzz) [inline server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss:155]: **nitpick:** Could you alphabetize these? And I think put the @include first?

**Review by dkatzz — COMMENTED** [review]
Updates look awesome!! Just some nits and a question about why we're removing questionAriaLabel.

File: server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss [hunk 2]
```diff
@@ -148,6 +165,34 @@ it is around the label */
...
 }
 
+.block-summary {
+  @include u-margin-top(4);
+  @include u-padding-y(2);
+}
+
+.block-summary-header {
+  display: flex;
+  align-items: flex-start; // Align items to the top
```
[the remark this answers is not in the document]
Comment (ssandbekkhaug) [reply server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss:175]: Done

File: server/app/views/applicant/QuestionSummaryFragment.html [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
+  <div>
+    <!-- Horizontal separator line -->
+    <div class="border-t border-base-lighter content"></div>
+
+    <div class="usa-prose container">
+      <!-- TODO(#8793): Add optional * indicator -->
+      <span
+        class="grid-col-6 left-align zero-margin mobile-truncate-3"
+        th:text="${answerData.questionText()}"
+      ></span>
```
[the remark this answers is not in the document]
Comment (ssandbekkhaug) [reply server/app/views/applicant/QuestionSummaryFragment.html:11]: The aria-label was mistakenly added here (by me in a previous PR). It reads "opens in a new window" even though it's not a link.

File: server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (dkatzz) [reply server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss:175]: @ssandbekkhaug I'm not seeing those updates - would you mind double checking?

File: server/app/views/applicant/QuestionSummaryFragment.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkatzz) [reply server/app/views/applicant/QuestionSummaryFragment.html:11]: Thank you! Could you add that info in the PR description?

**Review by dkatzz — APPROVED** [review]

File: server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (ssandbekkhaug) [reply server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss:174]: Sorry, it should be updated now

**Review by dkatzz — APPROVED** [review]

File: server/app/views/applicant/QuestionSummaryFragment.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (ssandbekkhaug) [reply server/app/views/applicant/QuestionSummaryFragment.html:11]: Done

**Pull request merged**